### PR TITLE
Add bootstrap icons

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Frontend

### Description
This PR adds [bootstrap icons](https://icons.getbootstrap.com/) to the frontend and removes fork awesome.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] Added / updated documentation
- [x] Added changelog snippet
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: frontend/src/components/history-page/sort-button/sort-button.tsx [hunk 1]
```diff
@@ -20,17 +22,8 @@ export enum SortModeEnum {
...
-      return 'sort'
-    case SortModeEnum.up:
-      return 'sort-asc'
-    case SortModeEnum.down:
-      return 'sort-desc'
-    default:
-      return 'sort'
-  }
+const getIcon = (direction: SortModeEnum): Icon => {
+  return direction === SortModeEnum.down ? IconSortAlphaDown : IconSortAlphaUp
```
Comment (DerMolly) [inline frontend/src/components/history-page/sort-button/sort-button.tsx:26]: SortModeEnum.up is not handled anymore?

File: frontend/src/components/markdown-renderer/extensions/emoji/emoji-markdown-extension.test.tsx [hunk 2]
```diff
@@ -24,10 +24,8 @@ describe('Emoji Markdown Extension', () => {
     expect(view.container).toMatchSnapshot()
   })
 
-  it('renders a fork awesome code', () => {
-    const view = render(
-      <TestMarkdownRenderer extensions={[new EmojiMarkdownExtension()]} content={':fa-circle-thin:'} />
-    )
+  it('renders an icon code', () => {
```
Comment (DerMolly) [inline frontend/src/components/markdown-renderer/extensions/emoji/emoji-markdown-extension.test.tsx:27]: This is not rendered. Either the test has the wrong name or the test does not work

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/hedge-doc-logo/hedge-doc-logo-with-text.tsx [hunk 3]
```diff
@@ -33,17 +32,13 @@ export enum HedgeDocLogoType {
  * @param logoType The logo type to be used.
  */
 export const HedgeDocLogoWithText: React.FC<HedgeDocLogoProps> = ({ size = HedgeDocLogoSize.MEDIUM, logoType }) => {
-  const { t } = useTranslation()
-  const altText = useMemo(() => t('app.icon'), [t])
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply frontend/src/components/common/hedge-doc-logo/hedge-doc-logo-with-text.tsx:37]: SVGs don't have an alt text or title

File: frontend/src/components/history-page/sort-button/sort-button.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply frontend/src/components/history-page/sort-button/sort-button.tsx:26]: It is. Not down => up

**Review by DerMolly — APPROVED** [review]

File: frontend/src/components/common/icons/bootstrap-icons.tsx [hunk 4]
```diff
@@ -0,0 +1,1994 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React from 'react'
+
+export type BootstrapIconName = keyof typeof BootstrapLazyIcons
+
+export const isBootstrapIconName = (name: string): name is BootstrapIconName => {
```
Comment (ErikMichelson) [inline frontend/src/components/common/icons/bootstrap-icons.tsx:10]: > `name is BootstrapIconName` 

Nice return type syntax 👍

File: frontend/src/components/history-page/sort-button/sort-button.tsx (hunk 1 above)
Comment (ErikMichelson) [reply frontend/src/components/history-page/sort-button/sort-button.tsx:26]: So there is no icon for "unsorted" anymore?

File: frontend/src/components/history-page/sort-button/sort-button.tsx [hunk 5]
```diff
@@ -64,13 +57,17 @@ const toggleDirection = (direction: SortModeEnum) => {
...
+export const SortButton: React.FC<SortButtonProps> = ({ children, onDirectionChange, direction }) => {
   const toggleSort = () => {
     onDirectionChange(toggleDirection(direction))
   }
 
   return (
-    <IconButton onClick={toggleSort} variant={variant} icon={getIcon(direction)} border={true}>
+    <IconButton
+      onClick={toggleSort}
+      variant={direction === SortModeEnum.no ? 'outline-light' : 'light'}
```
Comment (ErikMichelson) [inline frontend/src/components/history-page/sort-button/sort-button.tsx:68]: image

I'm not sure about this design decision yet. Isn't there any other possibility to indicate which sort button is active and which not?

File: frontend/src/components/history-page/sort-button/sort-button.tsx [hunk 6]
```diff
@@ -64,13 +57,17 @@ const toggleDirection = (direction: SortModeEnum) => {
...
+      icon={getIcon(direction)}
```
Comment (ErikMichelson) [inline frontend/src/components/history-page/sort-button/sort-button.tsx:69]: This is a function. Does React handle it properly or should we encapsulate it with a useMemo?

Comment (mrdrogdrog) [reply frontend/src/components/history-page/sort-button/sort-button.tsx:69]: not necessary.

File: frontend/src/components/history-page/sort-button/sort-button.tsx (hunk 5 above)
Comment (mrdrogdrog) [reply frontend/src/components/history-page/sort-button/sort-button.tsx]: I changed it

File: frontend/src/components/history-page/sort-button/sort-button.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply frontend/src/components/history-page/sort-button/sort-button.tsx]: It is now. I use the X icon.

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**